### PR TITLE
Testing WebRTC with iOS

### PR DESCRIPTION
**Pull request by herbertli**
Installed dependencies and got things compiling and running on my phone, however capturing audio and video doesn't seem to work.

### CODE REVIEW

File: config.xml [hunk 1]
```diff
@@ -107,11 +106,33 @@
...
+    <plugin name="cordova-plugin-geolocation" spec="^2.4.3">
+        <variable name="GEOLOCATION_USAGE_DESCRIPTION" value=" " />
+    </plugin>
+    <plugin name="cordova-plugin-globalization" spec="^1.0.9" />
+    <plugin name="cordova-plugin-inappbrowser" spec="^1.7.2" />
+    <plugin name="cordova-plugin-network-information" spec="^1.3.4" />
+    <plugin name="cordova-plugin-vibration" spec="^2.1.6" />
+    <plugin name="cordova-plugin-whitelist" spec="^1.3.3" />
     <plugin name="cordova-plugin-contacts" spec="^3.0.1">
-        <variable name="CONTACTS_USAGE_DESCRIPTION" value="Games Portal wants to use your contacts" />
```
Comment (yoav-zibin) [inline config.xml]: Please look at your changes before sending them: you're deleting code that YiWei wrote...

File: www/index.html [hunk 2]
```diff
@@ -55,6 +55,33 @@ <h1>PhoneGap</h1>
...
+      <video id="remotevideo" autoplay style="border: 5px solid black"/>
+    </span>
+    <span>
+      <h3>Local video</h3>
+      <video id="localvideo" autoplay style="border: 5px solid black"/>
+    </span>
+  
+    <script type="text/javascript" charset="utf-8" src="cordova.js"></script> 
+    <script src="https://www.gstatic.com/firebasejs/4.5.0/firebase.js"></script>
+    <script src="https://webrtc.github.io/adapter/adapter-latest.js"></script>
```
Comment (yoav-zibin) [inline www/index.html]: Use
cordova.plugins.iosrtc.registerGlobals
Now you have window.RTCPeerConnection, navigator.getUserMedia, etc.



See:
https://www.npmjs.com/package/cordova-plugin-iosrtc

// Just for Cordova apps. 
document.addEventListener('deviceready', function () {
  // Just for iOS devices. 
  if (window.device.platform === 'iOS') {
    cordova.plugins.iosrtc.registerGlobals();

**Review by yoav-zibin — COMMENTED** [review]
Can you email me a screenshot of what you see on the developer console in safari?
I'm sure you'll see some errors that we'll help you debug mistakes :)

**Comment by herbertli** [comment]
@yoav-zibin I got it working for iOS (with the updated example on the site [here](https://yoav-zibin.github.io/FirebaseExample/firebaseWebRTC.html)) basically every time you attach a stream to a video element you just need to call `cordova.plugins.iosrtc.observeVideo(video)`

File: config.xml [hunk 3]
```diff
@@ -107,11 +107,34 @@
         <allow-intent href="itms:*" />
         <allow-intent href="itms-apps:*" />
     </platform>
-    <plugin name="cordova-plugin-whitelist" spec="~1.3.3" />
```
Comment (yoav-zibin) [inline config.xml]: keep

File: config.xml [hunk 4]
```diff
@@ -107,11 +107,34 @@
...
+    <plugin name="cordova-plugin-geolocation" spec="^2.4.3">
+        <variable name="GEOLOCATION_USAGE_DESCRIPTION" value=" " />
+    </plugin>
+    <plugin name="cordova-plugin-globalization" spec="^1.0.9" />
+    <plugin name="cordova-plugin-inappbrowser" spec="^1.7.2" />
+    <plugin name="cordova-plugin-network-information" spec="^1.3.4" />
+    <plugin name="cordova-plugin-vibration" spec="^2.1.6" />
+    <plugin name="cordova-plugin-whitelist" spec="^1.3.3" />
     <plugin name="cordova-plugin-contacts" spec="^3.0.1">
-        <variable name="CONTACTS_USAGE_DESCRIPTION" value="Games Portal wants to use your contacts" />
```
Comment (yoav-zibin) [inline config.xml]: keep

File: package.json [hunk 5]
```diff
@@ -4,20 +4,61 @@
   "displayName": "Games Portal",
   "cordova": {
     "platforms": [
-      "android"
+      "android",
+      "ios"
     ],
     "plugins": {
       "cordova-plugin-contacts": {
-        "CONTACTS_USAGE_DESCRIPTION": "Games Portal wants to use your contacts"
```
Comment (yoav-zibin) [inline package.json]: keep

**Review by yoav-zibin — CHANGES_REQUESTED** [review]
Just careful with the files you share with YiWei so you don't mess up her contacts work  (please go through all of them and verify you don't mess it up)

File: package.json (hunk 5 above)
Comment (herbertli) [reply package.json]: My bad, I think when I added dependencies/did the setup for iOS, it over wrote some of hers, I'll add them back in!

**Review by yoav-zibin — APPROVED** [review]

**Pull request merged**